### ISSUE TITLE
Touch OSK blocks keys without text which may have rules behind

### ISSUE THREAD

**Issue by darcywong00** [keymanapp/keyman#249]
<a href="https://github.com/mcdurdin"><img src="https://avatars1.githubusercontent.com/u/4498365?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [mcdurdin](https://github.com/mcdurdin)**
_Tuesday Sept 23, 2014 at 19:53 GMT_
_Originally opened as KD-218_

----

If a key has no text on its cap, it is blocked by the following code in osk.clickKey:
```javascript
// Prevent any output from 'ghost' (unmapped) keys
if(keyName != 'K_SPACE') {
    var keyText=e.childNodes[0].innerHTML;
     if (keyText == '' || keyText == ' ') 
        return true; 
}
```

This is incorrect, because you may have, for instance, a special T_SPACE on a layer which has a + [T_SPACE] rule. I have this in the Lao Syllable keyboard.
If the key has no rule associated, and no key cap, then it won't do anything anyway?

**Comment by jahorton** [comment]
The issue with this being that we need to omit 'ghost' keys at present in order to remove unused layers later.  Though, that constraint may be going away with an upcoming redesign.

**Comment by jahorton** [comment]
This has been resolved as of #300 with the chirality support redesign, which additionally optimized how layers are defined such that the previously-mentioned detection is no longer required.
